### PR TITLE
Speed up search.

### PR DESCRIPTION
**Pull request by alien1976**
Tasks & test plan. - https://coherent-labs.atlassian.net/browse/COH-16850
https://coherent-labs.atlassian.net/browse/COH-16857

### CODE REVIEW

File: assets/js/fullSearch.js [hunk 1]
```diff
@@ -128,14 +129,43 @@ var pagination = document.getElementById('pagination');
...
+    pagination.innerHTML = '';
+    suggestions.classList.remove('loader-visible');
+    suggestions.classList.remove('d-none');
+  }
+
+  function addLoader() {
+    const loader = document.createElement('div');
+    loader.classList.add('loader');
+    suggestions.classList.add('loader-visible');
+    const text = document.createElement('span');
```
Comment (venetak) [inline assets/js/fullSearch.js]: Why do we create a new element each time. Wouldn't it be easier if we show/hide the element?

File: assets/js/search-utilities.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  try {
+    regex = new RegExp(regex, 'g');
+  } catch (error) {
+    return result;
+  }
+
+  while (match = regex.exec(str)) {
+    if (limit > -1 && i >= limit) break;
+    result.push(match.index);
+    i++
```
Comment (venetak) [inline assets/js/search-utilities.js]:
```suggestion
    i++;
```

**Review by Tony-CL — APPROVED** [review]
Searching/Loading visibility is excellent!

**Review by Tony-CL — APPROVED** [review]
We've tested this and works flawlessly and is visible enough for the user. Thank you for a proper test plan.

File: assets/js/fullSearch.js (hunk 1 above)
Comment (alien1976) [reply assets/js/fullSearch.js]: Made it to show/hide loader

File: assets/js/search-utilities.js (hunk 2 above)
Comment (alien1976) [reply assets/js/search-utilities.js]: Fixed

**Review by venetak — APPROVED** [review]

**Pull request merged**